userspace/libsinsp/event.cpp

```diff
@@ -1934,9 +1911,9 @@ const char* sinsp_evt::get_param_as_str(uint32_t id, OUT const char** resolved_s
 								 m_paramstr_storage.size(),
 								 "%s:%s->%s:%s",
 								 srcstr,
-								 port_to_string(*(uint16_t*)(payload + 17), (m_fdinfo != NULL) ? m_fdinfo->get_l4proto() : SCAP_L4_UNKNOWN, m_inspector->m_hostname_and_port_resolution_enabled).c_str(),
+								 port_to_string(*(uint16_t*)(param->m_val + 17), (m_fdinfo != NULL) ? m_fdinfo->get_l4proto() : SCAP_L4_UNKNOWN, m_inspector->m_hostname_and_port_resolution_enabled).c_str(),
```

Comment:
I am wondering, isn't this an UB too?
```
*(uint16_t*)(param->m_val + 17)
```

And same at the following line?

Reply:
Great Catch (TM)

Reply:
Fixed <3